### PR TITLE
[clang][Interp] Support __real/__imag on primitives

### PR DESCRIPTION
**Pull request by tbaederr**

### CODE REVIEW

File: clang/lib/AST/Interp/ByteCodeExprGen.cpp [hunk 1]
```diff
@@ -2761,6 +2761,8 @@ bool ByteCodeExprGen<Emitter>::VisitUnaryOperator(const UnaryOperator *E) {
       return false;
     return DiscardResult ? this->emitPop(*T, E) : this->emitComp(*T, E);
   case UO_Real: { // __real x
+    if (T)
+      return this->delegate(SubExpr);
     assert(!T);
```
Comment (cor3ntin) [inline clang/lib/AST/Interp/ByteCodeExprGen.cpp:2766]: Not sure the assert adds anything now

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Pull request merged**